### ISSUE TITLE
unexpected script compilation error

### ISSUE THREAD

**Issue by GDCCP** [nextflow-io/nextflow#1925]
## Bug report 

I try to test the 'scope' of variable with and without def. However, I run into an unexpected 'already defined variable' with this main.nf:

```nextflow
process p1 {
  publishDir path: "${workflow.launchDir}/sink"
  input:
    val(myfile)
  output:
    path("${myfile}.${delay}.txt")
  script:
    delay="${Math.abs(new Random().nextInt() % myfile.length()) + myfile.length()}"
    "sleep ${delay}; (echo ${delay} && echo ${myfile}) > ${myfile}.${delay}.txt"
}

process m2 {
  publishDir path: "${workflow.launchDir}/check"
  input:
    path(txt)
  output:
    path("${dest}")
  script:
    dest = "${txt}.copy"
    def delay="${Math.abs(new Random().nextInt() % dest.length()) + dest.length()}"
    "(sleep ${delay} && echo ${dest} ${delay} && cat ${txt} ${txt}) > ${dest}"
}

workflow {
  m2(p1(Channel.of('crazy','1','2','global','verylongstring',"xios",'3','last')))
}
```
### Expected behavior and actual behavior

Following [issue 678](https://github.com/nextflow-io/nextflow/issues/678), variable delay in m2 is defined with 'def'. However, it causes a script compilation error: 

> **Variable `dest` already defined in the process scope @ line 20, column 69.**

Note that it complains about dest, not delay. Removing the def does get rid of the error.

### Steps to reproduce the problem

I run nextflow 20.07.1 as follows:

nextflow run -dsl2 main.nf

### Environment 

* Nextflow version: [20.07.1] 
* Java version: [1.8.0]
* Operating system: [Centos 7]
* Bash version: (4.2.46)

**Comment by pditommaso** [comment]
Not properly formatting the code in the issue greatly reduce the likely to receive attention

**Comment by pditommaso** [comment]
The problem is caused by missing dsl=2 header 


```
nextflow.enable.dsl=2
```

tho, I agree the error message is confusing.

**Comment by GDCCP** [comment]
I don't think I need nextflow.enable.dsl=2 with the -dsl2 option. I indeed have been using -dsl2 for a while without nextflow.preview.dsl=2 or next.enable.dsl=2 for a while with 20.01.0 and 20.07.1.

I did add the enable line at the top of my main.nf and I still get the same error. This is what I see in the console with this command

`/software/Nextflow/20.07.1/bin/nextflow` run -dsl2 ./main.nf`

N E X T F L O W  ~  version 20.07.1
Launching `./main.nf` [sleepy_lovelace] - revision: 66c7a12935
Script compilation error
- file : /var/tmp/global-test/main.nf
- cause: Variable `dest` already defined in the process scope @ line 20, column 53.
   .abs(new Random().nextInt() % dest.lengt
                                 ^

Script compilation error
- file : /var/tmp/global-test/main.nf
- cause: Variable `dest` already defined in the process scope @ line 20, column 70.
   .nextInt() % dest.length()) + dest.lengt
                                 ^

2 errors
